### PR TITLE
[GOVUKAPP-767] Add all topics screen

### PR DESCRIPTION
**Pull request by 1pretz1**
## Video preview

https://github.com/user-attachments/assets/ab65abf3-604e-4e16-a7a5-e513b645d676

### CODE REVIEW

File: Production/govuk_ios/ViewControllers/AllTopicsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        return viewModel.topics.count
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let topic = viewModel.topics[indexPath.row]
+        viewModel.didTapTopic(topic)
+    }
+}
+
+class TopicTableViewCell: UITableViewCell {
```
Comment (batesyuk) [inline Production/govuk_ios/ViewControllers/AllTopicsViewController.swift:93]: Can you bang this in a new file, please?

File: Production/govuk_ios/ViewControllers/AllTopicsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        configureConstraints()
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+
+        self.topicCard.subviews.forEach({ $0.removeFromSuperview() })
+    }
+
+    func configureConstraints() {
```
Comment (batesyuk) [inline Production/govuk_ios/ViewControllers/AllTopicsViewController.swift:119]: private

File: Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import UIKit
+import Foundation
+
+class AllTopicsViewModel {
+    let analyticsService: AnalyticsServiceInterface
+    let topicAction: ((Topic) -> Void)?
+    let topics: [Topic]
+    var downloadError: TopicsListError?
```
Comment (batesyuk) [inline Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift:8]: Do all of these need to be internal? `private` or `private(set)` for some maybe?

**Review by batesyuk — CHANGES_REQUESTED** [review]

File: Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift (hunk 3 above)
Comment (1pretz1) [reply Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift]: They've changed a tad but have updated according to your comment

**Comment by 1pretz1** [comment]
Closing in favour of https://github.com/alphagov/govuk-mobile-ios-app/pull/137 (sonar scans seemingly broken on this PR)

**Pull request closed without merging**